File: TFG/ftp_tfg/lib/domain/usecases/auth/login_user.dart
import '../../../domain/entities/app_user.dart';
import '../../../domain/repositories/auth_repository.dart';

class LoginUser {
  final AuthRepository repository;

  LoginUser(this.repository);

  Future<AppUser> execute({
    required String email,
    required String password,
  }) =>
      repository.signInWithEmailAndPassword(
        email: email,
        password: password,
      );
}

File: TFG/hotftp_api/src/application/use-cases/auth/login-user.ts
import type { AppUser } from '../../../domain/entities/user.js';
import type { AuthRepository } from '../../../domain/repositories/auth-repository.js';

export class LoginUser {
  constructor(private readonly authRepository: AuthRepository) {}

  execute(email: string, password: string): Promise<AppUser> {
    return this.authRepository.login(email, password);
  }
}

